vpn/src/main/java/com/duckduckgo/mobile/android/vpn/processor/requestingapp/DetectOriginatingAppPackageModern.kt

```diff
@@ -41,8 +41,20 @@ class DetectOriginatingAppPackageModern(
     }
 
     private fun getPackageIdForUid(uid: Int): String {
-        return packageManager.getNameForUid(uid) ?: OriginatingAppPackageIdentifierStrategy.UNKNOWN.also {
-            Timber.i("Failed to get package ID for UID: $uid")
+        val packages = packageManager.getPackagesForUid(uid)
+        if (packages.isNullOrEmpty()) {
+            Timber.w("Failed to get package ID for UID: $uid")
+            return OriginatingAppPackageIdentifierStrategy.UNKNOWN
         }
+
+        if (packages.size > 1) {
```

Comment:
if there are multiple packages, that's _interesting_ so we log it in the console (to maybe notice it in development) but not so interesting that we'd want anything more at this stage (e.g., a pixel)

in any case, we'll grab the `first` package from the list if multiple are returned, so there is a slight chance we'll mis-identify one app for another but I think it's still worth the change until we see examples where this happens